ambry-frontend/src/main/java/com/github/ambry/frontend/FrontendRestRequestService.java

```diff
@@ -774,11 +777,11 @@ public void onCompletion(Void result, Exception exception) {
           headCallback.markStartTime();
           router.getBlob(convertedId, new GetBlobOptionsBuilder().operationType(GetBlobOptions.OperationType.BlobInfo)
               .getOption(getOption)
-              .build(), headCallback);
+              .build(), headCallback, QuotaChargeEventListener.buildQuotaChargeEventListener(restRequest, quotaManager, false));
           break;
         case DELETE:
           deleteCallback.markStartTime();
-          router.deleteBlob(convertedId, getHeader(restRequest.getArgs(), Headers.SERVICE_ID, false), deleteCallback);
+          router.deleteBlob(convertedId, getHeader(restRequest.getArgs(), Headers.SERVICE_ID, false), deleteCallback, QuotaChargeEventListener.buildQuotaChargeEventListener(restRequest, quotaManager, false));
```

Comment:
we probably don't need to do Chunk by Chunk on Delete.

Reply:
Although we are passing `QuotaChargeEventListener` to all methods of `Router`, we are not charging quota on chunk operation for every operation. In delete, ttlupdate and undelete operation, we are charging quota when the main blob operation is complete.